.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish Python 🐍 distributions 📦 to PyPI
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@master
+    
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
+    
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
+        verbose: true
```

Comment:
If we do trigger on a tag, I want two more steps following this one:

- If a release does not exist for that tag, create a new release named after the tag.
- Upload the distribution files to the release as well as pypi.

Reply:
^ This seems like a good idea regardless of the triggering mechanism. Can you put that in a separate issue for future work? (in the `main`-triggered configuration, we'd be pulling the version number for the release from the distribution info somehow, which at its core is defined in the setup.py file.)